cpp/oneapi/dal/table/backend/interop/block_info.hpp

```diff
@@ -0,0 +1,71 @@
+/*******************************************************************************
+* Copyright 2020-2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include <limits>
+#include <daal/include/data_management/data/numeric_table.h>
+
+#include "oneapi/dal/detail/common.hpp"
+
+namespace oneapi::dal::backend::interop {
+
+class block_info {
+    template <typename T>
+    using block_desc_t = daal::data_management::BlockDescriptor<T>;
+
+public:
+    template <typename BlockData>
+    block_info(const block_desc_t<BlockData>& block,
+               std::size_t row_begin_index,
+               std::size_t value_count,
+               std::size_t column_index)
+            : block_info(block, row_begin_index, value_count) {
+        this->column_index = detail::integral_cast<std::int64_t>(column_index);
+        this->single_column_requested = true;
```

Comment:
`block_info` is a simple service struct - what are the objections to make it immutable? 
Immutability makes every class harder to implement - and there are no guarantees it requires to be mutable in the future. It can be hard to make it mutable if we make it immutable now.
If you want to use the objects of that class like immutable ones - create `const` instances of it. It is much simpler and better than redesigning the class